src/neuro_flow/expr.py

```diff
@@ -1367,6 +1367,53 @@ class URIExpr(URIExprMixin, StrictExpr[URL]):
     pass
 
 
+class PlatformResourceURIExpr(URIExpr):
+    async def eval(self, root: RootABC) -> URL:
+        ret = await super().eval(root)
+        try:
+            # Hack to get flow's project ctx -> project_name
+            project_ctx = root.lookup("project")
+        except LookupError:
+            project_ctx = None
+
+        if project_ctx:
+            async with root.client() as cl:
+                ret = cl.parse.str_to_uri(
+                    str(ret),
+                    project_name=project_ctx.project_name,  # type: ignore
+                    short=True,
+                )
+        return ret
+
+
+class OptImageRefStrExpr(OptStrExpr):
+    async def eval(self, root: RootABC) -> Optional[str]:
+        ret = await super().eval(root)
+        project_ctx = None
+        if ret and ret.startswith("image:"):
+            try:
+                # Hack to get flow's project ctx -> project_name
+                project_ctx = root.lookup("project")
+            except LookupError:
+                pass
+        if ret and project_ctx:
+            async with root.client() as cl:
+                uri = cl.parse.str_to_uri(
+                    ret,
```

Comment:
do we need to cast to `str` here like we do in `PlatformResourceURIExpr.eval`?

Reply:
actually no, since the parent is OptStrExpr, so `ret = await super().eval(root)` will be `str | None`